src/core/namespace.h

```diff
@@ -93,11 +93,19 @@ struct TemporaryFileSystem {
         char *options;
 };
 
+typedef enum MountImageType {
+        MOUNT_IMAGE_DISCRETE,
+        MOUNT_IMAGE_OVERLAY,
```

Comment:
should be named MOUNT_IMAGE_EXTENSION, no? it doesn't do generic overlays after all, only our extensions

Reply:
I kinda went with "how it's used" rather than "what it implements", but changed as suggested